src/hybrid_failure_discovery/envs/conveyorbelt_env.py

```diff
@@ -0,0 +1,247 @@
+"""This module defines the ConveyorBeltEnv environment for simulating a 1D
+conveyor belt with loading and shifting actions."""
+
+from dataclasses import dataclass
+
+import matplotlib.pyplot as plt
+import numpy as np
+from gymnasium.core import RenderFrame
+from numpy.typing import NDArray
+from tomsutils.spaces import EnumSpace
+from tomsutils.utils import fig2data
+
+from hybrid_failure_discovery.envs.constraint_based_env_model import (
+    ConstraintBasedGymEnv,
+)
+
+
+@dataclass(frozen=True)
+class ConveyorBeltState:
+    """Represents the current state of the conveyor belt, including box
+    values."""
+
+    values: NDArray[np.float32]  # Vector of 6 float32s representing the state
+
+
+@dataclass(frozen=True)
+class ConveyorBeltAction:
+    """Represents a discrete action taken in the conveyor belt environment."""
+
+    index: int  # Discrete action index from 0 to 2
+
+
+class ConveyorBeltEnv(ConstraintBasedGymEnv[ConveyorBeltState, ConveyorBeltAction]):
+    """A simplified 1D conveyor belt environment with load/unload behavior and
+    3 actions."""
+
+    metadata = {"render_modes": ["rgb_array"], "render_fps": 10}
+
+    def __init__(self, render_mode: str = "rgb_array", seed: int = 0) -> None:
+        self.render_mode = render_mode
+        self.observation_dim = 6
+        self._last_inserted_value: float | None = None
+        self.action_dim = 3  # Now only actions 0, 1, 2 are valid
+        super().__init__(seed)
+
+        self._step_count = 0
+
+    def _create_action_space(self):
+        return range(self.action_dim)
+
+    def _get_obs(self) -> ConveyorBeltState:
+        assert self._current_state is not None
+        return self._current_state
+
+    def get_initial_states(self) -> EnumSpace[ConveyorBeltState]:
+        assert self._np_random is not None
+        init_values = self._np_random.random(self.observation_dim).astype(np.float32)
+        init_state = ConveyorBeltState(values=init_values)
+        return EnumSpace([init_state])
+
+    def get_next_states(
+        self, state: ConveyorBeltState, action: ConveyorBeltAction
+    ) -> EnumSpace[ConveyorBeltState]:
+        # print(f"[DEBUG] Action taken: {action.index}")
```

Comment:
I usually like to avoid commented out code to keep the main branch clean, but if these print statements will be helpful in the near term it's okay to leave them.